Standardise error messages

joyngjr opened the pull request:
### Summary:

Fixes #152 

Error message displayed for invalid repository names on initialisation and changing repositories are now the same. 

However, there are still some differences in behaviour between entering an invalid repository on initialisation and changing to an invalid repository. When an invalid repository name is entered on initialisation, an empty repository is created but the `label-filter-bar.component` is never loaded. On the other hand, the change to an invalid repo is rejected when users attempt to change to one, and the filter bar component doesn't seem to cause an issue. Perhaps it would be better to change the order of input validation in the future so an invalid repo can be flagged out at the login page, or we could create a default filter bar component to be loaded?

#### Type of change:
- 🐛 Bug Fix
- 🎨 Code Refactoring

### Changes Made:

- Refactor all error messages into `error-message.service.ts`
- Edit logger message in `phase.service.ts` to standardise format
- Add additional check for valid github repository when logging in for the first time before fetching data

### Screenshots:

https://github.com/CATcher-org/WATcher/assets/97519138/c9bbac02-b0a4-40e9-920e-89d4f4ce55a9


### Proposed Commit Message:

```
Standardise error messages for invalid repository names

Let's refactor all error messages to standardise them
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [ ] I have tested my changes thoroughly.
- [ ] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [X] I have added or modified code comments to improve code readability where necessary.
- [ ] I have updated the project's documentation as necessary.

</details>

joyngjr commented:
@luminousleek Hi, could I get your help to review this PR?

luminousleek commented on src/app/auth/session-selection/session-selection.component.ts:
```diff
@@ -85,6 +87,22 @@ export class SessionSelectionComponent implements OnInit {
     });
   }
 
+  /**
+   * Extracts the Organization Details from the input sessionInformation.
+   * @param sessionInformation - string in the format of 'orgName/dataRepo'
+   */
+  private getOrgDetails(sessionInformation: string) {
+    return sessionInformation.split('/')[0];
+  }
+
+  /**
+   * Extracts the Data Repository Details from the input sessionInformation.
+   * @param sessionInformation - string in the format of 'orgName/dataRepo'
+   */
+  private getDataRepoDetails(sessionInformation: string) {
+    return sessionInformation.split('/')[1];
+  }
+
```
Not sure what this is for since it's not used?

luminousleek commented on src/app/auth/session-selection/session-selection.component.ts:
```diff
@@ -31,6 +32,7 @@ export class SessionSelectionComponent implements OnInit {
     private authService: AuthService,
     private repoUrlCacheService: RepoUrlCacheService,
     private errorHandlingService: ErrorHandlingService,
+    private githubService: GithubService,
```
Not sure why this service is injected when it isn't used?

luminousleek reviewed (changes requested):
Generally looks good, thanks for refactoring all the error messages into a single service. 

I think you're trying to implement the extra check for repo validity in `session-selection.component.ts` now, but that should be in a separate PR.

There's also a bunch of linting changes but it looks ok to me. Maybe you can put that in your PR and commit message as well. 

Please also follow the commit message guidelines in https://se-education.org/guides/conventions/git.html

joyngjr commented:
@luminousleek Thanks for your help! I've changed back to double quotation marks since it doesn't make a difference, and removed the unused code from when I tried a different implementation.

luminousleek reviewed (approved):
LGTM, thanks for your work!

joyngjr commented:
@luminousleek I've modified the `codegen.yml` file to resolve merge conflicts. Could I get your help to review this PR again?

luminousleek reviewed (approved):
LGTM, sorry for the late review!
